contrib/babelfishpg_tsql/src/dbcmds.c

```
@@ -1372,15 +1396,20 @@ grant_perms_to_dbreader_dbwriter(const uint16 dbid,
 	namespace_rel_descr = RelationGetDescr(namespace_rel);
 
 	initStringInfo(&query);
-	appendStringInfo(&query, "GRANT SELECT ON ALL TABLES IN SCHEMA dummy TO dummy; ");
-	appendStringInfo(&query, "GRANT INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA dummy TO dummy; ");
+	appendStringInfo(&query, "GRANT SELECT ON ALL TABLES IN SCHEMA dummy TO dummy; "); /* db_reader */
+	appendStringInfo(&query, "GRANT INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA dummy TO dummy; "); /* db_writer */
+	appendStringInfo(&query, "GRANT TRUNCATE ON ALL TABLES IN SCHEMA dummy TO dummy; "); /* db_ddladmin */
+	appendStringInfo(&query, "GRANT CREATE ON SCHEMA dummy TO dummy ; "); /* db_ddladmin */
 
 	/* Grant ALTER DEFAULT PRIVILEGES on schema owner and dbo user. */
-	appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy, dummy IN SCHEMA dummy GRANT SELECT ON TABLES TO dummy; ");
-	appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy, dummy IN SCHEMA dummy GRANT INSERT, UPDATE, DELETE ON TABLES TO dummy; ");
+	appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy, dummy IN SCHEMA dummy GRANT SELECT ON TABLES TO dummy; "); /*db_reader */
+	appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy, dummy IN SCHEMA dummy GRANT INSERT, UPDATE, DELETE ON TABLES TO dummy; "); /* db_writer */
+	appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy, dummy IN SCHEMA dummy GRANT TRUNCATE ON TABLES TO dummy; "); /* db_ddladmin */
 
 	stmt_list = raw_parser(query.data, RAW_PARSE_DEFAULT);
 
+	Assert(list_length(stmt_list) == 7);
```

Should we throw the error? As discussed in earlier comments since it's better to error out than crashing the server.

---

Done